app/controllers/make_a_demo_payment/confirm_controller.js

```diff
@@ -11,41 +11,37 @@ const publicAuthClient = require('../../services/clients/public_auth_client')
 const auth = require('../../services/auth_service.js')
 
 module.exports = (req, res) => {
-  let protoData = lodash.get(req, 'session.pageData.protoData')
-  let gatewayAccountId = auth.getCurrentGatewayAccountId(req)
+  const gatewayAccountId = auth.getCurrentGatewayAccountId(req)
+  const {paymentAmount, paymentDescription} = lodash.get(req, 'session.pageData.makeADemoPayment')
 
-  if (protoData) {
-    let params = {
-      indexPage: paths.user.loggedIn,
-      name: protoData.protoPaymentDescription,
-      price: protoData.protoPaymentAmount,
-      gateway_account_id: gatewayAccountId
-    }
+  if (!paymentAmount || !paymentDescription) {
+    return res.redirect(paths.prototyping.demoPayment.index)
+  }
 
-    publicAuthClient.createTokenForAccount({
-      accountId: gatewayAccountId,
-      correlationId: req.correlationId,
-      payload: {
-        account_id: gatewayAccountId,
-        created_by: req.user.email,
-        description: `Token for Prototype: ${req.body['payment-description']}`
-      }
-    })
-      .then(publicAuthData => productsClient.product.create({
-        payApiToken: publicAuthData.token,
-        gatewayAccountId,
-        name: protoData.protoPaymentDescription,
-        price: protoData.protoPaymentAmount
-      }))
-      .then(product => {
-        params.prototypeLink = lodash.get(product, 'links.pay.href')
-        return response(req, res, 'dashboard/demo-payment/confirm', params)
-      })
-      .catch(error => {
-        req.flash('genericError', `<h2>There were errors</h2> ${error}`)
-        return res.redirect(paths.prototyping.demoPayment.index)
+  publicAuthClient.createTokenForAccount({
+    accountId: gatewayAccountId,
+    correlationId: req.correlationId,
+    payload: {
+      account_id: gatewayAccountId,
+      created_by: req.user.email,
+      description: `Token for Demo Payment`
+    }
+  })
+    .then(publicAuthData => productsClient.product.create({
+      payApiToken: publicAuthData.token,
+      gatewayAccountId,
+      name: paymentDescription,
+      price: paymentAmount * 100
```

Comment:
Isn't it safer to use a `Math.trunc()` or `Math.floor()` around the calculated payment amount. irrespective of what validations were taken care of by the caller modules.

Reply:
Yeah, sorry raised the PR as a WIP. Was about to add this.

Reply:
Actually I'm not sure it is, going by our existing validation model. 

If we we're to do this in the Node.js code, surely we should be doing this at the ProductsClient level. But by that logic, surely we should also be validating that it also a Number.

As this api expects an Integer for price, shouldn't the API return 400 if it's passed a float?